.github/workflows/gohai.yml

```diff
@@ -0,0 +1,26 @@
+name: "Gohai Test"
+
+# Only run the tests if pkg/gohai was changed
+on:
+  push:
+    paths:
+      - 'pkg/gohai/**'
+  pull_request:
+    paths:
+      - 'pkg/gohai/**'
+jobs:
+  gohai_test:
+    strategy:
+      matrix:
+        # Use oldest and latest available ubuntu, macos and windows
+        os: [ubuntu-20.04, ubuntu-latest, macos-11, macos-latest, windows-2019, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v3
+      # we don't want to just use the agent's version because gohai might be used outside of it
+      # eg. opentelemetry
+      with:
+        go-version: 1.17.x
```

Comment:
We may want to have the tests run with the same Golang version that the Agent builds with since it may be even more important that it works in the Agent itself than as an import.

Reply:
Following our conversation offline I updated the config to run with both the agent and gohai's go versions.